### PR TITLE
Magic Login: Use Jetpack Colours

### LINKED ISSUE #33781: Jetpack connection: "email login link" page doesn't use Jetpack colors
During Jetpack onboarding flow, the dark Jetpack color scheme is dropped out for "email me login link" -page:

image

image

It should stay consistent across the flow.

See paObgF-fe-p2

### PR DESCRIPTION
**Pull request by Aurorum**
#### Changes proposed in this Pull Request

* During the Jetpack Connect flow, clicking "Email me a login link" results in the colours for the flow being dropped. This PR proposes changing that.

#### Testing instructions

1. Open an incognito tab and try to link a Jetpack site to WordPress.com
2. When asked to login, click "Email me a login link"
3. Verify you're redirected to `log-in/jetpack/link`

Screenshot 2019-07-28 at 15 21 39

Screenshot 2019-07-28 at 15 22 47

Verify that "Enter a password" and "Back" redirects back to the Jetpack login page. Then visit `/log-in` and verify that nothing there has changed including the links at the bottom, and that `/log-in/link` doesn't have the Jetpack logo or colours.

This does not change the behaviour from after the emailed link is clicked, including the colours after the link is expired. I think that it's not possible in a PR to add `jetpack` to the link in the email sent. Also, signing in doesn't redirect back to the Jetpack login page (it redirects to the Reader), but this PR only intends to change the colours.

cc @simison 

Fixes #33781

### CODE REVIEW

**Review by jeffgolenski — APPROVED** [review]

**Comment by Aurorum** [comment]
Resolved conflicts. :)

**Comment by arunsathiya** [comment]
@Aurorum, I am seeing some lint failures with:

```
'gridicons' import is restricted from being used. Please use 'components/gridicon' instead.
```

Would this be something you can update?

@jeffgolenski @Automattic/luna, are we still planning on landing this, when @Aurorum fix the lint errors? Thanks!

**Comment by Aurorum** [comment]
Yep, done! 👍

**Comment by Aurorum** [comment]
I clearly did something wrong... 

> ModuleNotFoundError: Module not found: Error: Can't resolve 'components/gridicons' in '/home/circleci/wp-calypso/client/login/magic-login

I'll look over this tomorrow.

**Comment by Aurorum** [comment]
@arunsathiya I can't seem to get it working unfortunately. :(

I've reverted the initial changes I made, and even though the linter is failing, can we leave fixing the Gridicons issue for another PR? It isn't an issue introduced in this PR, and doesn't seem like a simple fix.

**Comment by arunsathiya** [comment]
It looks like the component name is `gridicon` and not `gridicons`. So, import should be `components/gridicon`.

**Comment by Aurorum** [comment]
Oh, oops... 

Thanks, that's turned the tests green!

**Comment by Aurorum** [comment]
Conflicts resolved :)

**Comment by simison** [comment]
You might need to rebase with this since SVN imports were little broken: https://github.com/Automattic/wp-calypso/pull/36139

File: client/login/wp-login/style.scss [hunk 1]
```diff
@@ -191,7 +192,8 @@ $image-height: 47px;
 	.wp-login__links a,
 	.logged-out-form__links a,
 	.logged-out-form__link-item,
-	.translator-invite__content a {
+	.translator-invite__content a,
+  	.magic-login__footer a {
```
Comment (simison) [inline client/login/wp-login/style.scss]: Small fix needed; there's a mixture of tabs and spaces here.

Comment (Aurorum) [reply client/login/wp-login/style.scss]: Fixed the formatting of that file 👍

**Comment by Aurorum** [comment]
I'll take a look at fixing this PR soon, I've currently been trying to wrestle getting Calypso working again locally.

Screenshot 2019-09-15 at 20 54 15

Once I manage to do that, I'll try to get this PR in a mergeable state. :)

**Comment by simison** [comment]
@Aurorum have you tried removing `node_modules` folder completely and then `npm start` again?

**Comment by simison** [comment]
This is testing pretty well. I found an issue which is present also in `master` so no worries in this PR:

if I click steps _"Email me a login link → Enter a password instead → Create a new account"_ I end up in "add new site" screen instead of "create an account":

Screenshot 2019-09-15 at 16 16 01
Screenshot 2019-09-15 at 16 15 42

I think it's a matter of different URL attributes getting lost when clicking between different steps.

Error component doesn't look great on dark background but that's not a blocker for this PR and  already an issue in other similar screens:

Screenshot 2019-09-15 at 16 09 54

Since this is little riskier PR (as it touches very critical paths), I'll leave deploying this for some time next week when I'm back in the office. :-)

**Review by simison — APPROVED** [review]

**Comment by Aurorum** [comment]
Sadly got the same error after trying that, probably will take some more Googling to get it working again

**Comment by Aurorum** [comment]
Conflicts resolved. :)

**Comment by bluefuton** [comment]
@simison are you happy to 🚢 this one?

**Review by tyxla — APPROVED** [review]
This works well and code looks great! Tested default and Jetpack-branded logins, plus some related functionality, everything works well! Great work!

Ideally, we could do some improvements in subsequent PRs:

* Having controller set a context param for jetpack login, and using that later instead of working with the raw route.
* Persisting Jetpack login in the redirect URL that is sent with the magic login email.
* Adapting login page that user lands on from the magic login link to be Jetpack-branded.

But these are all subsequent work and some of them might require API changes.

This piece makes the entire experience much better than it was already, and I'm super happy to see it! Thanks a bunch @Aurorum for your contribution! 🙌

**Pull request merged**